coordinator/src/services/translation/request/dispense/dispense-claim.ts

```diff
@@ -0,0 +1,305 @@
+  }
+
+  const hl7PriorPrescriptionReleaseEventRef = createPriorPrescriptionReleaseEventRef(messageHeader)
+  dispenseClaim.sequelTo = new hl7V3.SequelTo(hl7PriorPrescriptionReleaseEventRef)
+
+  return dispenseClaim
+}
+
+function isExemption(chargeExemptionCode: string) {
+  //TODO - create enum?
+  return chargeExemptionCode !== "0001"
+}
+
+function isEvidenceSeen(evidenceSeenCode: string) {
+  //TODO - create enum?
+  return evidenceSeenCode === "evidence-seen"
+}
+
+async function createDispenseClaimPertinentInformation1(
+  claim: fhir.Claim,
+  item: fhir.ClaimItem,
+  messageId: string,
+  timestamp: hl7V3.Timestamp,
+  logger: pino.Logger
+) {
+  //TODO - work out what this means and whether we're doing it:
+  // "Note: this must refer to the last one in the series if more
+  // than one dispense event was required to fulfil the prescription."
+  const supplyHeader = new hl7V3.DispenseClaimSupplyHeader(new hl7V3.GlobalIdentifier(messageId))
+
+  //TODO - repeat dispensing
+
+  const payeeOdsCode = claim.payee.party.identifier.value
+  //TODO - check that we can omit the user details (applies to all dispensing messages)
+  const agentPerson = await createAgentPersonForUnattendedAccess(payeeOdsCode, logger)
+  supplyHeader.legalAuthenticator = new hl7V3.LegalAuthenticator(timestamp, agentPerson)
+
+  //TODO - populate pertinentInformation2 (non-dispensing reason)
+
+  const prescriptionStatus = createPrescriptionStatus(item)
+  supplyHeader.pertinentInformation3 = new hl7V3.SupplyHeaderPertinentInformation3(prescriptionStatus)
+
+  supplyHeader.pertinentInformation1 = item.detail.map(detail => {
+    const suppliedLineItem = createSuppliedLineItem(claim, item, detail)
+    return new hl7V3.SupplyHeaderPertinentInformation1(suppliedLineItem)
+  })
+
+  const prescriptionId = createPrescriptionId(claim)
+  supplyHeader.pertinentInformation4 = new hl7V3.SupplyHeaderPertinentInformation4(prescriptionId)
+
+  const originalPrescriptionRef = createOriginalPrescriptionRef(claim)
+  supplyHeader.inFulfillmentOf = new hl7V3.InFulfillmentOf(originalPrescriptionRef)
+
+  return new hl7V3.DispenseCommonPertinentInformation1(supplyHeader)
+}
+
+function createPrescriptionStatus(item: fhir.ClaimItem) {
+  const prescriptionStatusExtension = getExtensionForUrl(
+    item.extension,
+    "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus",
+    "Claim.item.extension"
+  ) as fhir.CodingExtension
+  const prescriptionStatusCoding = prescriptionStatusExtension.valueCoding
+  return new hl7V3.PrescriptionStatus(prescriptionStatusCoding.code, prescriptionStatusCoding.display)
+}
+
+export function createSuppliedLineItem(
+  claim: fhir.Claim,
+  item: fhir.ClaimItem,
+  detail: fhir.ClaimItemDetail
+): hl7V3.DispenseClaimSuppliedLineItem {
+  const claimSequenceIdentifierExtension = getExtensionForUrl(
+    detail.extension,
+    "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimSequenceIdentifier",
```

Comment:
this identifier might have to reference the line item id

Reply:
Asking BSA